rest/perf_routes.go

```diff
@@ -203,7 +203,8 @@ func (h *perfCountByTaskIdHandler) Parse(_ context.Context, r *http.Request) err
 // number of PerformanceResults found
 func (h *perfCountByTaskIdHandler) Run(ctx context.Context) gimlet.Responder {
 	perfResults, err := h.sc.FindPerformanceResults(ctx, h.opts)
-	if err != nil {
+	errResp, ok := err.(gimlet.ErrorResponse)
+	if err != nil && (!ok || errResp.StatusCode != http.StatusNotFound) {
```

Comment:
I would push this logic down to the `data` package.

Reply:
I think pushing the logic down would require modifying the other routes that also depend on the 404 to handle the "no results" case. Is it okay if the other routes (e.g. `perfGetByTaskNameHandler`) return an empty response when there's no results?

In any case, I think it's probably better to not 404 in the case of no results since a 404 usually means that the REST route is not a valid route, not that it returned empty results. But I don't know if any users are depending on 404 meaning "no results" instead of "no such REST route".

Reply:
@julianedwards what do you think about Kim's arguments.

Reply:
Ok that makes sense, I am fine with this then.